shm/Cargo.hpp

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <string>
+
+class Cargo {
+public:
+    Cargo(const std::string& name,
+          size_t amount,
+          size_t basePrice)
+        : _name(name), _amount(amount), _basePrice(basePrice) {}

```

Comment:
only definition of trivial getter/ setter could be in .hpp. Please move other to .cpp file.